internal/database/user_credentials.go

```diff
@@ -95,19 +147,24 @@ type UserCredentialScope struct {
 // Create creates a new user credential based on the given scope and
 // authenticator. If the scope already has a credential, an error will be
 // returned.
-func (s *UserCredentialsStore) Create(ctx context.Context, scope UserCredentialScope, credential auth.Authenticator) (*UserCredential, error) {
+func (s *UserCredentialsStore) Create(ctx context.Context, enc encryption.Encrypter, scope UserCredentialScope, credential auth.Authenticator) (*UserCredential, error) {
 	if Mocks.UserCredentials.Create != nil {
-		return Mocks.UserCredentials.Create(ctx, scope, credential)
+		return Mocks.UserCredentials.Create(ctx, enc, scope, credential)
 	}
 	s.ensureStore()
 
+	secret, err := encryptAuthenticator(ctx, enc, credential)
```

Comment:
Nitpick but I'm an encryption doofus who regularily mixes up encryption words and if I read "secret" I think "key" or "encryption secret" but not "encrypted and thus secret data".

Feel free to ignore though.

```suggestion
	encryptedAuthenticator, err := encryptAuthenticator(ctx, enc, credential)
```

Reply:
Good call. I used a different name in the end, but I did change it.